src/Campaigns/Actions/AssignDuplicatedFormToCampaign.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Campaigns\Repositories\CampaignRepository;
+use Give\Log\Log;
+
+/**
+ * @unreleased
+ */
+class AssignDuplicatedFormToCampaign
+{
+    /**
+     * @unreleased
+     * @var CampaignRepository
+     */
+    protected $campaignRepository;
+
+    /**
+     * @unreleased
+     */
+    public function __construct(CampaignRepository $campaignRepository)
+    {
+        $this->campaignRepository = $campaignRepository;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function __invoke($duplicatedFormID, $originalFormID)
+    {
+        $campaign = $this->campaignRepository->queryByFormId($originalFormID)->get();
+
+        try {
+            $this->campaignRepository->addCampaignForm($campaign, $duplicatedFormID, true);
```

Comment:
Since `$campaign` is nullable, I would recommend checking if it exists first to avoid the potential of a fatal error from the typed param of `addCampaignForm`
<img width="457" alt="Screenshot 2025-02-07 at 3 02 38 PM" src="https://github.com/user-attachments/assets/d1a65f72-a902-4657-a0a3-482175854004" />

Reply:
Good point. I added a check and a corresponding test, see [4bcf91a](https://github.com/impress-org/givewp/pull/7712/commits/4bcf91a021e6b8229ad95be767999514540d8b18).

Reply:
Thanks!